docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,120 @@
+Frontend
+~~~~~~~~
 
-State-driven applications
-=========================
+We recommended building your frontend with the `React <https://reactjs.org>`_ framework. We provide
+several support libraries to build your React frontend for a DAML application.
 
-**State-driven** applications build up a real-time view of the ledger state by reading events and recording contract create and archive events. They then generate commands based on a given state, not just single events.
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| Name                                                         | Summary                                                                  |
++==============================================================+==========================================================================+
+| `@daml/react <https://www.npmjs.com/package/@daml/react>`_   | React hooks to query/create/exercise DAML contracts                      |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/ledger <https://www.npmjs.com/package/@daml/ledger>`_ | DAML ledger object to connect and directly submit commands to the ledger |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/types  <https://www.npmjs.com/package/@daml/types>`_  | DAML data types                                                          |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
 
-Examples of these are automation and interactive applications that let a user or code react to complex state on the ledger (e.g. the DA Navigator tool).
+You can install any of these libraries by running ``yarn add @daml/react`` in the ``ui`` directory
+of your project. Please explore the ``create-daml-app`` example project to see the usage of these
+libraries.
 
-Which approach to take
-======================
+The interface between the frontend and the backend of your application is defined by your DAML code.
+To make your life easy when interacting with the ledger, the DAML assistant can generate
+corresponding typescript data definitions for the data types declared in DAML.
 
-For all except the simplest applications, we generally recommend the state-driven approach. State-driven applications are easier to reason about when determining correctness, so this makes design and implementation easier.
+.. code-block:: bash
 
-In practice, most applications are actually a mixture of the two styles, with one predominating. It is easier to add some event handling to a state-driven application, so it is better to start with that style.
+  daml codegen ts .daml/dist/<your-project-name.dar> -o daml2ts
```

Comment:
We need to mention what's a dar file and reference the appropriate section in the docs, e.g., /concepts/glossary.html#dar-file-dalf-file

Reply:
I think that's more of a detail that you don't necessarily need to know at this point. I think it's more confusing than enlightening if you just want to get started.